src/hdf5_time_series_storage.jl

```diff
@@ -182,43 +200,78 @@ function remove_time_series!(
     end
 end
 
+function _make_time_array(timestamps, data, columns::Union{Vector{Symbol}, Nothing})
+    if columns === nothing
+        return TimeSeries.TimeArray(timestamps, data)
+    else
+        return TimeSeries.TimeArray(timestamps, data, columns)
+    end
+end
+
 function get_time_series(
     storage::Hdf5TimeSeriesStorage,
-    uuid::UUIDs.UUID;
-    index = 0,
-    len = 0,
-)::TimeSeries.TimeArray
+    uuid::UUIDs.UUID,
+    index::Int,
+    len::Int,
+    count::Int,
+)::DataStructures.SortedDict{Dates.DateTime, TimeSeries.TimeArray}
     return HDF5.h5open(storage.file_path, "r") do file
         root = _get_root(storage, file)
         path = _get_time_series_path(root, uuid)
-
-        if index == 0
-            data = HDF5.read(path["data"])
-            timestamps = HDF5.read(path["timestamps"])
+        _initial_time_stamp = HDF5.read(HDF5.attrs(path)["initial_time"])
+        initial_time_stamp = Dates.epochms2datetime(_initial_time_stamp)
+        resolution = Dates.Millisecond(HDF5.read(HDF5.attrs(path)["resolution"]))
+        series_length = HDF5.read(HDF5.attrs(path)["length"])
+        if HDF5.exists(HDF5.attrs(path), "columns")
+            columns = Symbol.(HDF5.read(HDF5.attrs(path)["columns"]))
         else
-            @assert len != 0
-            end_index = index + len - 1
-            # HDF5.readmmap could be faster in many cases than this. However, experiments
-            # resulted in various crashes if we tried to close the file before references
-            # to the array data were garbage collected. May need to consult with the
-            # Julia HDF5 library maintainers about that.
-            sz_tuple = size(path["data"])
-            if length(sz_tuple) == 1
+            columns = nothing
+        end
+
+        # HDF5.readmmap could be faster in many cases than this. However, experiments
+        # resulted in various crashes if we tried to close the file before references
+        # to the array data were garbage collected. May need to consult with the
+        # Julia HDF5 library maintainers about that.
+        # TODO: Consider case Interval_ms = 0 if contigous instead of not existing
+        if !HDF5.exists(HDF5.attrs(path), "interval")
+            @assert HDF5.read(HDF5.attrs(path)["count"]) == 1
+            @debug "reconstructing a contigouos time series"
+            time_stamps =
+                range(initial_time_stamp; length = series_length, step = resolution)
+            if len == series_length && index == 1
+                data = HDF5.read(path["data"])
+            elseif len <= series_length
+                end_index = min(index + len - 1, series_length)
```

Comment:
1. This allows the user to pass an invalid valid for `len`.  That will just lead to more problems for them.  We should throw an error in that case.
2. I suspect that this is an off-by-one error.  `index + len`

Reply:
Checked, no off-by-one error. Agreed on the improvement of conditional